Gems/Atom/Tools/ShaderManagementConsole/Code/Source/Document/ShaderManagementConsoleDocument.cpp

```diff
@@ -60,6 +60,33 @@ namespace ShaderManagementConsole
         ShaderManagementConsoleDocumentRequestBus::Handler::BusDisconnect();
     }
 
+    AZ::u32 ShaderManagementConsoleDocument::AddOneVariantRow()
+    {
+        auto shaderAssetResult =
+            AZ::RPI::AssetUtils::LoadAsset<AZ::RPI::ShaderAsset>(m_absolutePath,
+                                                                 m_shaderVariantListSourceData.m_shaderFilePath);
+        if (shaderAssetResult)
+        {
+            BeginEdit();
+            AZ::Data::Asset<AZ::RPI::ShaderAsset> shaderAsset = shaderAssetResult.GetValue();
+            AZStd::vector<AZ::RPI::ShaderOptionDescriptor> options = shaderAsset->GetShaderOptionGroupLayout()->GetShaderOptions();
+            AZ::RPI::ShaderVariantListSourceData::VariantInfo variantInfo;
+            for (auto& optionDescriptor : options)
+            {
+                variantInfo.m_options[optionDescriptor.GetName()] = ""; // all unset
+            }
+            variantInfo.m_stableId = m_shaderVariantListSourceData.m_shaderVariants.empty()
+                ? 0
+                : m_shaderVariantListSourceData.m_shaderVariants.back().m_stableId + 1;
+            m_shaderVariantListSourceData.m_shaderVariants.push_back(variantInfo);
+            EndEdit();
+            AtomToolsFramework::AtomToolsDocumentNotificationBus::Event(
+                m_toolId, &AtomToolsFramework::AtomToolsDocumentNotificationBus::Events::OnDocumentModified, m_id);
+            return variantInfo.m_stableId;
+        }
+        return {};
+    }
+
```

Comment:
Refer to SetShaderVariantListSourceData to see all of the notifications that must be sent in order to update all of the observers and the inspector. This will prevent the crash that was demonstrated earlier.

This function could be simplified by checking to see if the document is open, copying the shader variant list source data, making any modifications, then calling SetShaderVariantListSourceData to update the data with all of the correct notifications. 

I would recommend moving the begin edit and end edit to the call site instead of calling them from within the function. That would allow the function to be called more than once as part of a batch between begin edit and edit calls.

This function should also be added to the shader management console document request bus and the behavior context for this class.